app/Jobs/FetchAd.php

```diff
@@ -37,28 +45,27 @@ public function __construct(string $sourceName)
      * @param AdItemInserter $adItemInserter
      * @param AdPicTransferrer $adPicTransferrer
      */
-    public function handle(AdFetcher $adFetcher , AdItemInserter $adItemInserter , AdPicTransferrer $adPicTransferrer)
+    public function handle(AdFetcher $adFetcher, AdItemInserter $adItemInserter, AdPicTransferrer $adPicTransferrer)
     {
         $this->adFetcher = $adFetcher;
-        $this->adItemInserter=$adItemInserter;
-        $this->adPicTransferrer=$adPicTransferrer;
+        $this->adItemInserter = $adItemInserter;
+        $this->adPicTransferrer = $adPicTransferrer;
 
         /** @var stdClass $source */
-        $source = Repo::getRecords('sources' , ['*'], ['name'=>$this->sourceName])->first();
-        if(isset($source)){
-            [$donePages, $failedPages] = $this->fetch($source);
+        if (isset($this->source)) {
+            [$donePages, $failedPages] = $this->fetch($this->source, $this->since);
```

Comment:
remove $source and since from fetch method signature.

Reply:
done